src/ui/class-user-interface.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Duplicate Post user interface.
+ *
+ * @package Duplicate_Post
+ */
+
+namespace Yoast\WP\Duplicate_Post\UI;
+
+use Yoast\WP\Duplicate_Post\Permissions_Helper;
+
+/**
+ * Represents the Duplicate Post User Interface class.
+ */
+class User_Interface {
+
+	/**
+	 * Holds the permissions helper.
+	 *
+	 * @var Permissions_Helper
+	 */
+	protected $permissions_helper;
+
+	/**
+	 * Holds the object to manage the row actions for the post.
+	 *
+	 * @var Row_Actions
+	 */
+	protected $row_actions;
+
+	/**
+	 * Holds the object to manage the post submitbox links.
+	 *
+	 * @var Post_Submitbox
+	 */
+	protected $post_submitbox;
+
+	/**
+	 * Holds the object to manage the block editor links.
+	 *
+	 * @var Block_Editor
+	 */
+	protected $block_editor;
+
+	/**
+	 * Holds the object to manage the admin bar links.
+	 *
+	 * @var Admin_Bar
+	 */
+	protected $admin_bar;
+
+	/**
+	 * Holds the object to manage the bulk actions dropdown.
+	 *
+	 * @var Bulk_Actions
+	 */
+	protected $bulk_actions;
+
+	/**
+	 * Admin notices object.
+	 *
+	 * @var Admin_Notices
+	 */
+	protected $admin_notices;
+
+	/**
+	 * Post states object.
+	 *
+	 * @var Post_States
+	 */
+	protected $post_states;
+
+	/**
+	 * Metabox object.
+	 *
+	 * @var Metabox
+	 */
+	protected $metabox;
+
+	/**
+	 * Column object.
+	 *
+	 * @var Column
+	 */
+	protected $column;
+
+	/**
+	 * Holds the object to create the action link to duplicate.
+	 *
+	 * @var Link_Builder
+	 */
+	protected $link_builder;
+
+	/**
+	 * Initializes the class.
+	 *
+	 * @param Permissions_Helper $permissions_helper The permissions helper object.
+	 */
+	public function __construct( Permissions_Helper $permissions_helper ) {
+		$this->permissions_helper = $permissions_helper;
+		$this->link_builder       = new Link_Builder();
+		$this->row_actions        = new Row_Actions( $this->link_builder, $this->permissions_helper );
+		$this->post_submitbox     = new Post_Submitbox( $this->link_builder, $this->permissions_helper );
+		$this->block_editor       = new Block_Editor( $this->link_builder, $this->permissions_helper );
+		$this->admin_bar          = new Admin_Bar( $this->link_builder, $this->permissions_helper );
+		$this->bulk_actions       = new Bulk_Actions( $this->permissions_helper );
+		$this->admin_notices      = new Admin_Notices();
+		$this->post_states        = new Post_States();
+		$this->metabox            = new Metabox( $this->permissions_helper );
+		$this->column             = new Column( $this->permissions_helper );
```

Comment:
Not too sure what all these instances are doing here, as they don't seem to be used within the class itself.

Reply:
I could have created them in the main class, yes, but since the UI is now completely decoupled from the rest of the plugin (except for the actions mechanism of WordPress, and of course the permissions) it seemed natural and clearer to me to keep them distinct from the other non-UI classes, except for the Permissions Helper which is injected. This way we could theoretically completely rewrite the UI without changing anything outside the `ui` folder/`UI` namespace.

Reply:
So basically this is code that is preemptively being placed within this class then?